micropip/transaction.py

```diff
@@ -355,24 +363,31 @@ def eval_marker(e: dict[str, str]) -> bool:
             else:
                 return
 
-        if self.check_version_satisfied(req):
-            # Maybe while we were downloading pypi_json some other branch
-            # installed the wheel?
+        # Maybe while we were downloading pypi_json some other branch
+        # installed the wheel?
+        satisfied, ver = self.check_version_satisfied(req)
+        if satisfied:
+            logger.info(f"Requirement already satisfied: {req} ({ver})")
             return
 
-        await self.add_wheel(wheel, req.extras)
+        await self.add_wheel(wheel, req.extras, specifier=str(req.specifier))
 
     async def add_wheel(
         self,
         wheel: WheelInfo,
         extras: set[str],
+        *,
+        specifier: str = "",
```

Comment:
Not very clear to me out of context what the specifier is supposed to be. Can you maybe add a docstring for it (I imagine it's documented in req.specifier)?

Reply:
Added docstring for that method. Actually that parameter is used only for logging so I guess we need to remove that in the future.